doc/examples: golang - list_nodes.go

dougsland opened the pull request:
Just an example how to use golang and bluechi.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

Yarboa committed:
Fix due to aboot machine  tests

Changing to containers.kill instead of containers.stop python api
It reduce failures during cleanup.
containers killed instead of stop+wait option exist in API.

Updated test timeouts to complete success in slow OS.

Signed-off-by: Yariv Rachmani <yrachman@redhat.com>

Yarboa committed:
Changing Containers.kill -> back to Containers.stop(**)

It seens that the default wait adds 10 sec to each
container stop command,

from podman stop --helo
 -t, --time uint
 Seconds to wait for stop before killing the container (default 10)

Signed-off-by: Yariv Rachmani <yrachman@redhat.com>

engelmi committed:
added integration test for getting a unit property

Relates to https://github.com/containers/bluechi/issues/401
Adds an integration test that verifies that a property of
a systemd unit can be retrieved.

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
added integration test for getting all properties of a unit

Relates to https://github.com/containers/bluechi/issues/401
Adds an integration test that verifies that all properties of
a systemd unit can be retrieved.

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
added integration test for setting a unit property value

Relates to https://github.com/containers/bluechi/issues/401
Adds an integration test that verifies that a unit property value
of a systemd unit can be set.

Signed-off-by: Michael Engel <mengel@redhat.com>

dougsland committed:
markdownlint: ignore README.developer.md

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
README.developer: add busctl info for bluechi

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

dougsland committed:
doc/examples: golang - list_nodes.go

Just an example how to use golang and bluechi.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

engelmi commented on doc/bluechi-examples/golang/.gitignore:
```diff
@@ -0,0 +1 @@
+list_nodes
```
The `.gitignore` here is not needed. Could you please remove it?

engelmi commented on doc/bluechi-examples/golang/README.md:
```diff
@@ -0,0 +1,35 @@
+# golang examples
```
The content of this file should be appended to https://github.com/containers/bluechi/blob/main/doc/docs/api/examples.md as a new section
```
## GoLang

...
```

engelmi commented on doc/bluechi-examples/golang/list_nodes.go:
```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
```
The `doc/bluechi-examples` is intended for the bindings (only python currently) - maybe the directory name is misleading. Could you please move it to `doc/api-examples/go/list_nodes.go`?

engelmi commented on doc/bluechi-examples/golang/README.md:
````diff
@@ -0,0 +1,35 @@
+``` bash
+# go install github.com/godbus/dbus
+```
+
````
Could you add another section here for showing the content of `list_nodes.go`? Then you can integrate the source file like this
```go
--8<-- "api-examples/go/list_nodes.go"
```

Maybe we need to add a new package to the readthedocs, but we can do that after this PR is merged.

engelmi reviewed (changes requested):
I really like adding more languages to the list of examples on how to use the D-Bus API :) I have a few change requests, though.

engelmi committed:
added integration test for bluechi with invalid values of the port

Relates to https://github.com/containers/bluechi/issues/416
Adds an integration test that verifies BlueChi handles
invalid port values properly.

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
fix flaky bluechi-agent invalid port test

Signed-off-by: Michael Engel <mengel@redhat.com>

engelmi committed:
added integration test for running bluechi-agent alongside bluechi

Fixes https://github.com/containers/bluechi/issues/525
Adds an integration test that verifies that bluechi-agent can run
alongside bluechi controller on the same machine.

Signed-off-by: Michael Engel <mengel@redhat.com>
